packages/style-engine/class-wp-style-engine.php

```diff
@@ -30,6 +30,14 @@ class WP_Style_Engine {
 	 */
 	private static $instance = null;
 
+	/**
+	 * A white list of CSS custom properties.
+	 * Used to bypass safecss_filter_attr().
+	 */
+	const VALID_CUSTOM_PROPERTIES = array(
+		'--wp--style--block-gap' => array( 'spacing', 'blockGap' ),
```

Comment:
We have to do this because `safecss_filter_attr` (kses.php) doesn't allow CSS custom properties.